### PR TITLE
feat: unify the fields for sparse and dense array

### LINKED ISSUE #2438: Unify the naming and type for the `shape` field in DenseNdArray and SparseNdArray
**Describe the feature**
In `jina.proto`, the shape field in DenseNdArray and SparseNdArray are naming differently and using different types. 

```
message DenseNdArrayProto {
...
    // the shape (dimensions) of the array
    repeated uint32 shape = 2;
...
}

message SparseNdArrayProto {
...
    // A 1-D int64 tensor of shape [ndims], which specifies the dense_shape of the sparse tensor.
    repeated int64 dense_shape = 3;
}
```

**Your proposal**

- [ ] unify the type to  `repeated int64`
- [ ] unify the name to `shape`

---
<!-- Optional, but really help us locate the problem faster --
**Environment**
- jina 1.2.0
- jina-proto 0.0.81
- jina-vcs-tag (unset)
- libzmq 4.3.4
- pyzmq 1.20.3
- protobuf 3.17.0
- proto-backend cpp
- grpcio 1.37.1
- pyyaml 5.4.1
- python 3.8.5
- platform Darwin
- platform-release 20.3.0
- platform-version Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021;
  root:xnu-7195.81.3~1/RELEASE_X86_64
- architecture x86_64
- processor i386
- jina-resources /Users/nanwang/.pyenv/versions/3.8.5/lib/python3.8/site-
  packages/jina/resources
* JINA_ARRAY_QUANT (unset)
* JINA_BINARY_DELIMITER (unset)
* JINA_CONTRIB_MODULE (unset)
* JINA_CONTRIB_MODULE_IS_LOADING(unset)
* JINA_CONTROL_PORT (unset)
* JINA_DEFAULT_HOST (unset)
* JINA_DISABLE_UVLOOP (unset)
* JINA_EXECUTOR_WORKDIR (unset)
* JINA_FULL_CLI (unset)
* JINA_IPC_SOCK_TMP (unset)
* JINA_LOG_CONFIG (unset)
* JINA_LOG_ID (unset)
* JINA_LOG_LEVEL (unset)
* JINA_LOG_NO_COLOR (unset)
* JINA_LOG_WORKSPACE (unset)
* JINA_POD_NAME (unset)
* JINA_RAISE_ERROR_EARLY (unset)
* JINA_RANDOM_PORTS (unset)
* JINA_RANDOM_PORT_MAX (unset)
* JINA_RANDOM_PORT_MIN (unset)
* JINA_SOCKET_HWM (unset)
* JINA_VCS_VERSION (unset)
* JINA_WARN_UNNAMED (unset)
* JINA_WORKSPACE (unset)

**Screenshots**
image

### PR DESCRIPTION
**Pull request by nan-wang**
resolved #2438

### CODE REVIEW

**Review by niuzs-alan — APPROVED** [review]
LGTM

**Comment by niuzs-alan** [comment]
> LGTM
> 
> @nan-wang plz pull the latest change on the master to the feature branch

Done

**Comment by niuzs-alan** [comment]
@nan-wang Something new is happening.

**Comment by bwanglzu** [comment]
this change will affect some of our core code:

9F83E460-8368-44BC-BFAC-4E0604F16668

**Comment by bwanglzu** [comment]
we might also want to unify primitive types:

3C2EDF41-A6FA-4FD2-88D5-4560678D4A70

**Comment by bwanglzu** [comment]
@nan-wang should be ready

**Review by 0969397110 — DISMISSED** [review]

**Comment by nan-wang** [comment]
@bwanglzu Have you rerun the `docker run -v $(pwd)/jina/proto:/jina/proto jinaai/protogen` and bumped the `__proto_version__` in `jina/__init__.py`?

**Comment by bwanglzu** [comment]
@nan-wang I runned the docker command as was specified in `build-proto.sh`, I'm using mac, do I need to manually update the version number `__proto_version__`?

File: tests/unit/peapods/runtimes/asyncio/rest/test_models.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def test_jina_document_to_pydantic_document_sparse():
             == pydantic_doc.embedding.sparse.values.buffer.decode()
         )
         assert (
-            jina_doc['embedding']['sparse']['values']['shape']
+            list(map(int, jina_doc['embedding']['sparse']['values']['shape']))
```
Comment (JoanFM) [inline tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: why is this needed?

Comment (bwanglzu) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: @JoanFM after type changed, the `shape` is str, while the shape passed to pandantic is int

Comment (hanxiao) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: why `shape` is in str?

Comment (JoanFM) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: yes,  why is that?

Comment (JoanFM) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: the shape in proto is a repeated int64

Comment (bwanglzu) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: @JoanFM @hanxiao 
> As per proto3 JSON spec, uint64/int64 fields should be printed as decimal strings. See:
https://developers.google.com/protocol-buffers/docs/proto3#json

The reason is that uint64/int64 is not part of JSON spec and many JSON libraries only support double precision. To prevent precision loss our proto3 JSON spec requires serializers to put int64/uint64 values in strings.

checkout [here](https://github.com/protocolbuffers/protobuf/issues/2954#issuecomment-292671661)

Comment (bwanglzu) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: another option is to use `int32` to represent the `shape`, how do you think @nan-wang ?

Comment (hanxiao) [reply tests/unit/peapods/runtimes/asyncio/rest/test_models.py:256]: uint32 it is

**Comment by bwanglzu** [comment]
unified all `shape` to `uint32`

**Review by JoanFM — APPROVED** [review]

**Review by hanxiao — APPROVED** [review]
